### ISSUE TITLE
No rustdoc rendering warnings for Markdown files

### ISSUE THREAD

**Issue by ollie27** [rust-lang/rust#45365]
Running `rustdoc foo.md` on the following for example doesn't produce any warnings about the differences between hoedown and pulldown-cmark even if passing `--enable-commonmark`:

```markdown
# test

2^32
```

cc. #44229, @GuillaumeGomez
